File: sejeong/전화번호목록.py
def solution(phone_book):
    phone_book.sort()  # 전화번호를 정렬한후
    for i in range(len(phone_book)-1):
        if phone_book[i+1].startswith(phone_book[i]): # 그다음전화번호가 앞의번호로 시작하는지 확인
            return False
    return True


phone_book = ["119", "97674223", "1195524421"]
print(solution(phone_book))

File: subin/programmers/전화번호_목록.java
package programmers;

import java.util.Arrays;

class Phonebook {
    // 전화번호 목록을 입력받아 특정 조건을 확인한 후 true 또는 false를 반환하는 메서드
    public static boolean solution(String[] phone_book) {
        Arrays.sort(phone_book);
        int len = phone_book.length;

        for (int i = 0; i < len - 1; i++) {
            String prefix = phone_book[i];
            String next = phone_book[i + 1];
            // startsWith은 문자열이 특정 문자열로 시작하는지 확인하는 메서드
            // 지정된 접두사로 시작하면 true, 그렇지 않으면 false를 반환
            if (next.startsWith(prefix)) {
                return false;
            }
        }

        return true;
    }

    // 프로그램 실행을 위한 메인 메서드
    public static void main(String[] args) {
        System.out.println(solution(new String[]{"119", "97674223", "1195524421"}));
    }
}
